### PR TITLE
ptl-tool: allow not pushing branch to ceph-ci

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
Tested this change, works fine for me.

**Comment by rishabh-d-dave** [comment]
@batrick Is there anybody else who should be added as the reviewer? Can't think of anybody.

**Comment by rishabh-d-dave** [comment]
Tested it, it works fine.

File: src/script/ptl-tool.py [hunk 1]
```diff
@@ -548,6 +549,8 @@ def main():
     parser.add_argument('--qa-tags', dest='qa_tags', action='store', help='QA tags for tracker')
     parser.add_argument('--stop-at-built', dest='stop_at_built', action='store_true', help='stop execution when branch is built')
     parser.add_argument('--update-qa', dest='update_qa', action='store', help='update QA run ticket')
+    parser.add_argument('--no-build', dest='no_build', action='store_true',
```
Comment (batrick) [inline src/script/ptl-tool.py:552]:
```suggestion
    parser.add_argument('--no-push-ci', dest='no_push_ci', action='store_true',
```

File: src/script/ptl-tool.py [hunk 2]
```diff
@@ -477,8 +477,9 @@ def build_branch(args):
         if args.qa_tags:
             custom_fields.append({'id': REDMINE_CUSTOM_FIELD_ID_QA_TAGS, 'value': args.qa_tags})
 
-        G.git.push(CI_REMOTE_URL, branch) # for shaman
-        G.git.push(CI_REMOTE_URL, tag.name) # for archival
+        if not args.no_build:
+            G.git.push(CI_REMOTE_URL, branch) # for shaman
+            G.git.push(CI_REMOTE_URL, tag.name) # for archival
```
Comment (batrick) [inline src/script/ptl-tool.py:482]: In what circumstances do you find this useful?

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/script/ptl-tool.py (hunk 2 above)
Comment (rishabh-d-dave) [reply src/script/ptl-tool.py:482]: When I need to test a bunch of QA PRs urgently, I can do so by merging these PRs on a new branch and pass it to `--suite-branch` option while I use builds from shaman for recent version of main branch.

I've been doing this today lots of times. Venky wanted me to have a tracker ticket for future reference.

This change will be handy in future too when QA PRs needs to be collected and tested against main, squid, reef, etc.

File: src/script/ptl-tool.py (hunk 1 above)
Comment (rishabh-d-dave) [reply src/script/ptl-tool.py]: Done.

File: src/script/ptl-tool.py (hunk 2 above)
Comment (rishabh-d-dave) [reply src/script/ptl-tool.py:482]: I've mentioned this in Slack chat and commit message too. Let me know if it needs to be updaed.

Comment (batrick) [reply src/script/ptl-tool.py:482]: It's a fine use-case. I suggest adding it to the top-level README comments as an example. Be aware however that other changes in the qa suite on `--base main` (the default) may unexpectedly change the `qa/` suite in ways that surprise.

Comment (rishabh-d-dave) [reply src/script/ptl-tool.py:497]: Done.

File: src/script/ptl-tool.py [hunk 3]
```diff
@@ -79,6 +79,22 @@
 # ...
 #
 #
+# Merging all PRs labeled 'wip-pdonnell-testing' into a new test branch but
+# NOT pushing that branch to ceph-ci repo:
+#
+# $ src/script/ptl-tool.py --pr-label wip-pdonnell-testing --branch main --no-build
```
Comment (batrick) [inline src/script/ptl-tool.py:85]:
```suggestion
# $ src/script/ptl-tool.py --pr-label wip-pdonnell-testing --branch main --create-qa --no-push-ci
```

Comment (batrick) [reply src/script/ptl-tool.py:85]: It bears mentioning that we only push to ci if creating or updating a qa ticket.

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/script/ptl-tool.py (hunk 3 above)
Comment (rishabh-d-dave) [reply src/script/ptl-tool.py:86]: Okay, added a mention for this.

**Review by batrick — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-windows-pull-requests/40348/

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-windows-pull-requests/40359/

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-windows-pull-requests/40377/console
```
Build step 'Execute shell' marked build as failure
Archiving artifacts
[PostBuildScript] - [INFO] Executing post build scripts.
[ceph-windows-pull-requests] $ /bin/bash /tmp/jenkins4399865873181019505.sh
+ set -o errexit
+ set -o pipefail
+ delete_libvirt_vms
++ sudo virsh list --all --name
+ libvirt_vms='ceph-ubuntu-vstart-ceph-windows-pull-requests-40337
ceph-windows-client-ceph-windows-pull-requests-40337'
+ for vm in $libvirt_vms
+ sudo virsh destroy ceph-ubuntu-vstart-ceph-windows-pull-requests-40337
Domain 'ceph-ubuntu-vstart-ceph-windows-pull-requests-40337' destroyed

+ sudo virsh managedsave-remove ceph-ubuntu-vstart-ceph-windows-pull-requests-40337
Domain 'ceph-ubuntu-vstart-ceph-windows-pull-requests-40337' has no manage save image; removal skipped
+ sudo virsh undefine ceph-ubuntu-vstart-ceph-windows-pull-requests-40337
Domain 'ceph-ubuntu-vstart-ceph-windows-pull-requests-40337' has been undefined

+ for vm in $libvirt_vms
+ sudo virsh destroy ceph-windows-client-ceph-windows-pull-requests-40337
Domain 'ceph-windows-client-ceph-windows-pull-requests-40337' destroyed

+ sudo virsh managedsave-remove ceph-windows-client-ceph-windows-pull-requests-40337
Domain 'ceph-windows-client-ceph-windows-pull-requests-40337' has no manage save image; removal skipped
+ sudo virsh undefine ceph-windows-client-ceph-windows-pull-requests-40337
Domain 'ceph-windows-client-ceph-windows-pull-requests-40337' has been undefined

+ sudo find /var/lib/libvirt/images/ -type f -delete
+ sudo virsh pool-refresh default
error: failed to get pool 'default'
error: Storage pool not found: no storage pool with matching name 'default'

+ true
+ clear_libvirt_networks
++ sudo virsh net-list --all --name
+ networks=default
+ for network in $networks
+ sudo virsh net-destroy default
Network default destroyed

+ sudo virsh net-undefine default
Network default has been undefined

+ sudo rm -rf /home/jenkins-build/build/workspace/ceph-windows-pull-requests/ceph /home/jenkins-build/build/workspace/ceph-windows-pull-requests/ceph_vstart /home/jenkins-build/build/workspace/ceph-windows-pull-requests/ceph.zip /home/jenkins-build/build/workspace/ceph-windows-pull-requests/libvirt
[Checks API] No suitable checks publisher found.
Setting status of f40abdde1de8687b910d5b42ecc18a7333f86aff to FAILURE with url https://jenkins.ceph.com/job/ceph-windows-pull-requests/40377/ and message: 'ceph windows tests failed
 '
Using context: ceph windows tests
Finished: FAILURE
```

**Pull request merged**